app/views.py

```diff
@@ -1,131 +1,134 @@
+        ).first()]
+  elif subsection_slug == "main":
+    section =  models.Section.query.filter(
+      models.Section.slug == section_slug
+        ).first()
+    articles = models.Article.query.filter(
+      models.Article.section == section
+        ).all()
+  else:
+    subsection =  models.Subsection.query.filter(
+      models.Subsection.slug == subsection_slug
+        ).first()
+    articles = models.Article.query.filter(
+      models.Article.subsection == subsection
+        ).all()
+  secure_articles = []
+  for article in articles:
+    article_dict = {
+    "content": article.content, 
+    "datetime": article.datetime, 
+    "id": article.id, 
+    "isDraft": article.isDraft, 
+    "issue": article.issue, 
+    "section_id": article.section_id, 
+    "slug": article.slug, 
+    "subsection_id": article.subsection_id, 
+    "title": article.title, 
+    "volume": article.volume
+          }   
+    secure_articles.append(article_dict)
+  return jsonify({"articles": secure_articles})
 
-@app.route('/get_section_articles/sections/<string:section_slug>/subsection/<string:subsection_slug>/articles/', methods=['GET'], defaults={'article_slug': None}) 
-@app.route('/get_section_articles/sections/<string:section_slug>/subsection/<string:subsection_slug>/articles/<string:article_slug>', methods=['GET']) 
-def articles_in_section(section_slug,subsection_slug,article_slug):
-    if article_slug != None and article_slug != "None":
-        articles = [models.Article.query.filter(
-            models.Article.slug == article_slug
-                ).first()]
-    elif subsection_slug == "main":
-        section =  models.Section.query.filter(
-            models.Section.slug == section_slug
-                ).first()
-        articles = models.Article.query.filter(
-            models.Article.section == section
-                ).all()
-    else:
-        subsection =  models.Subsection.query.filter(
-            models.Subsection.slug == subsection_slug
-                ).first()
-        articles = models.Article.query.filter(
-            models.Article.subsection == subsection
-                ).all()
-    secure_articles = []
-    for article in articles:
-        article_dict = {
-      "content": article.content, 
-      "datetime": article.datetime, 
-      "id": article.id, 
-      "isDraft": article.isDraft, 
-      "issue": article.issue, 
-      "section_id": article.section_id, 
-      "slug": article.slug, 
-      "subsection_id": article.subsection_id, 
-      "title": article.title, 
-      "volume": article.volume
-                    }   
-        secure_articles.append(article_dict)
-    return jsonify({"articles": secure_articles})
 
+#----------------------------------------------  
 
-#----------------------------------------------    
-
-@app.route('/newspaper/<int:volume>/<int:issue>', methods=['GET'])
+@app.route('/newspaper/<int:volume>/<int:issue>' )
 def retrieve_article_data(volume,issue):
```

Comment:
this function name is ambiguous. specify in the function name how this article is being gotten.